src/sinks/appsignal/config.rs

```diff
@@ -0,0 +1,199 @@
+use futures::FutureExt;
+use http::{header::AUTHORIZATION, Request, Uri};
+use hyper::Body;
+use tower::ServiceBuilder;
+use vector_common::sensitive_string::SensitiveString;
+use vector_config::configurable_component;
+use vector_core::{
+    config::{proxy::ProxyConfig, AcknowledgementsConfig, DataType, Input},
+    tls::{MaybeTlsSettings, TlsEnableableConfig},
+};
+
+use crate::{
+    codecs::Transformer,
+    http::HttpClient,
+    sinks::{
+        prelude::{SinkConfig, SinkContext},
+        util::{
+            http::HttpStatusRetryLogic, BatchConfig, Compression, ServiceBuilderExt,
+            SinkBatchSettings, TowerRequestConfig,
+        },
+        BuildError, Healthcheck, HealthcheckError, VectorSink,
+    },
+};
+
+use super::{
+    service::{AppsignalResponse, AppsignalService},
+    sink::AppsignalSink,
+};
+
+/// Configuration for the `appsignal` sink.
+#[configurable_component(sink("appsignal", "Deliver log and metric event data to AppSignal."))]
+#[derive(Clone, Debug, Default)]
+pub struct AppsignalConfig {
+    /// The URI for the AppSignal API to send data to.
+    #[configurable(validation(format = "uri"))]
+    #[configurable(metadata(docs::examples = "https://appsignal-endpoint.net"))]
+    #[serde(default = "default_endpoint")]
+    pub endpoint: String,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏does this one need to be `pub` ? It should suffice to use pub(super), if it does, I would expect.

Reply:
It's set by the integration tests file so it needs to be accessible from there. `pub(super)` is better 👍